chore: lint fail on warnings

mjkeaton opened the pull request:
Before this PR, lint warnings did not fail the build and commits were accepted.
After this PR is merged, warnings will be treated as errors and precommit checks decline to commit.

There were two minor warnings:
```
./src/components/ui/form.tsx
  168:3  warning  Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components  react-refresh/only-export-components

./src/context/LanguageContext.tsx
  15:27  warning  Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components  react-refresh/only-export-components
```

`LanguageContext` has been split up into two files as suggested ("Move your React context(s) to a separate file  react-refresh/only-export-components"). In `form.tsx` the rule is ignored as the functionality is not used yet and it needs to be checked for a false positives and/or split up once it is actually in use (added a `TODO`).

## Test
Remove the ignore rule in `form.tsx` and try to commit the changes. Notice that it won't be allowed as the warning is treated as an error.

mjkeaton commented:
Still not sure why the warning is triggered in the first place. eslint config contains `{ allowConstantExport: true }` and Vite should handle it [as stated in eslint-plugin-react-refresh docs](https://github.com/ArnaudBarre/eslint-plugin-react-refresh?tab=readme-ov-file#allowconstantexport-v040).

cryptosalomao reviewed (changes requested):
LGTM. Just a suggestion: put all the language context files into a single directory

mjkeaton commented:
> LGTM. Just a suggestion: put all the language context files into a single directory

Done! Moved to directory `src/context/language/` and added basic tests.
